e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,186 @@
+	Link:        "https://access.redhat.com/errata/RHSA-2022:7089",
+	MetadataV2: &v1.Metadata{
+		PublishedDateTime:    "2022-10-24T00:00Z",
+		LastModifiedDateTime: "",
+		CvssV2:               nil,
+		CvssV3: &v1.CVSSMetadata{
+			Score:               8.1,
+			Vector:              "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
+			ExploitabilityScore: 2.2,
+			ImpactScore:         5.9,
+		},
+	},
+	FixedBy:  "0:1.3.5-8.el8_6",
+	Severity: "Important",
+}
+
+// vulnTar is an example of a non-fixable vulnerability
+var vulnTar = &v1.Vulnerability{
+	Name:        "CVE-2005-2541",
+	Description: "DOCUMENTATION: The MITRE CVE dictionary describes this issue as: Tar 1.15.1 does not properly warn the user when extracting setuid or setgid files, which may allow local users or remote attackers to gain privileges. \n            STATEMENT: This CVE was assigned to what is the documented and expected behaviour of tar.  There are currently no plans to change tar behaviour to strip setuid and setgid bits when extracting archives.",
+	Link:        "https://access.redhat.com/security/cve/CVE-2005-2541",
+	MetadataV2: &v1.Metadata{
+		CvssV2: nil,
+		CvssV3: &v1.CVSSMetadata{
+			Score:               7,
+			Vector:              "CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H",
+			ExploitabilityScore: 1,
+			ImpactScore:         5.9,
+		},
+	},
+	FixedBy:  "",
+	Severity: "Moderate",
+}
+
+func buildRequestCase(notes []v1.Note) *v1.GetNodeVulnerabilitiesRequest {
+	// From: https://www.redhat.com/security/data/metrics/repository-to-cpe.json
+	// "rhel-8-for-x86_64-appstream-rpms": {"cpes": ["cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream"]},
+	// "rhel-8-for-x86_64-baseos-rpms": {"cpes": ["cpe:/o:redhat:enterprise_linux:8::baseos", "cpe:/o:redhat:rhel:8.3::baseos"]}
+	cpes := []string{
+		"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+		"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+	}
+	return &v1.GetNodeVulnerabilitiesRequest{
+		OsImage:          "Red Hat Enterprise Linux CoreOS 45.82.202008101249-0 (Ootpa)",
+		KernelVersion:    "0.0.1",
+		KubeletVersion:   "0.0.1",
+		KubeproxyVersion: "0.0.1",
+		Runtime: &v1.GetNodeVulnerabilitiesRequest_ContainerRuntime{
+			Name:    "docker",
+			Version: "0.0.1",
+		},
+		Notes: notes,
+		Components: &v1.Components{
+			Namespace: "rhcos:4.11",
+			RhelComponents: []*v1.RHELComponent{
+				{
+					Id:        int64(1),
+					Name:      "libksba",
+					Namespace: "rhel:8",
```

Comment:
`Namespace` at the `Components` level is different from `Namespace` at the `RHELComponent` level. Which one is used by Scanner when it also sees `v1.Note_CERTIFIED_RHEL_SCAN_UNAVAILABLE` note?

Reply:
We seem to use `req.GetComponents()..GetNamespace()` inside of `func GetVulnerabilitiesForComponents` but it is used only to set the field inside of the `layers` struct and it does not seem to be used for scanning.

Reply:
How does scanner know that it is a RHEL-based OS and produces no features when you set `Note_CERTIFIED_RHEL_SCAN_UNAVAILABLE` note?

Reply:
The note `Note_CERTIFIED_RHEL_SCAN_UNAVAILABLE` is defined at the level of `GetNodeVulnerabilitiesRequest`:

```go
	// Handle the new format of the request and scan node inventory additionally
	if req.GetComponents() != nil {
		layer, err := apiV1.GetVulnerabilitiesForComponents(s.db, req.GetComponents(), common.HasUncertifiedRHEL(req.GetNotes()))
```

The behavior of Scanner in the scope of this API seem to not rely on the value of `Components.Namespace` - I confirmed that by using several random arbitrary chosen values of namespaces and was able to obtain the same scan results.

Regarding the `Components.RHELComponent[].Namespace`: the entire `Components.RHELComponent` is passed deeper into scanner code and I have not analyzed whether the behavior depends on the namespace value, however a quick experiment with inputs:

```
RHELComponent.Namespace: "xxx:8",
RHELComponent.Namespace: "yyy:8",
RHELComponent.Namespace: "xxx:999",
RHELComponent.Namespace: "xxx:1",
```

produces the same scan results as for `RHELComponent.Namespace: "rhel:8"`. The information about namespace (which is just copied from the input to the output) returned from `apiV1.GetVulnerabilitiesForComponents` is lost in `features.ConvertFeatures`, so we cannot even add an assertion for its value in this test.

Reply:
Interesting. This means that namespace doesn't play role in rhel vulnerabilities detection. Looks like only CPEs matter. No CPEs => no vulnerabilities.

Reply:
> No CPEs => no vulnerabilities

Yes, this is what I observed experimentally.